src/views/CountryNsOverviewContextAndStructure/NationalSocietyKeyDocuments/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { useOutletContext } from 'react-router-dom';
+import { DownloadFillIcon, SearchLineIcon } from '@ifrc-go/icons';
+import { listToGroupList, mapToList } from '@togglecorp/fujs';
+
+import Container from '#components/Container';
+import DateInput from '#components/DateInput';
+import TextInput from '#components/TextInput';
+import Link from '#components/Link';
+import useTranslation from '#hooks/useTranslation';
+import useFilterState from '#hooks/useFilterState';
+import { GoApiResponse, useRequest } from '#utils/restRequest';
+import { CountryOutletContext } from '#utils/outletContext';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type GetKeyDocumentResponse = GoApiResponse<'/api/v2/country-document/'>;
+export type KeyDocumentItem = NonNullable<GetKeyDocumentResponse['results']>[number];
+
+function NationalSocietyKeyDocuments() {
+    const strings = useTranslation(i18n);
+
+    const { countryId } = useOutletContext<CountryOutletContext>();
+    const {
+        filter,
+        rawFilter,
+        setFilterField,
+    } = useFilterState<{
+        searchText?: string,
+        startDateAfter?: string,
+        startDateBefore?: string,
+    }>({
+        filter: {},
+    });
+    const {
+        response: documentResponse,
+    } = useRequest({
+        url: '/api/v2/country-document/',
+        query: {
+            country: Number(countryId),
+            search: filter.searchText,
+            year__lte: filter.startDateAfter,
+            year__gte: filter.startDateBefore,
+        },
+        preserveResponse: true,
+    });
```

Comment:
we should check if `countryId` is defined. 
```suggestion
        skip: isNotDefined(countryId),
        url: '/api/v2/country-document/',
        query: {
            country: isDefined(countryId) ? Number(countryId) : undefined,
            search: filter.searchText,
            year__lte: filter.startDateAfter,
            year__gte: filter.startDateBefore,
        },
        preserveResponse: true,
    });
```